comp/core/tagger/tagger.go

```diff
@@ -0,0 +1,362 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package tagger
+
+import (
+	"context"
+	"reflect"
+	"sync"
+
+	configComponent "github.com/DataDog/datadog-agent/comp/core/config"
+	logComp "github.com/DataDog/datadog-agent/comp/core/log"
+	tagger_api "github.com/DataDog/datadog-agent/comp/core/tagger/api"
+	"github.com/DataDog/datadog-agent/comp/core/tagger/collectors"
+	"github.com/DataDog/datadog-agent/comp/core/tagger/local"
+	"github.com/DataDog/datadog-agent/comp/core/tagger/remote"
+	"github.com/DataDog/datadog-agent/comp/core/tagger/replay"
+	"github.com/DataDog/datadog-agent/comp/core/tagger/types"
+	"github.com/DataDog/datadog-agent/comp/core/tagger/utils"
+	"github.com/DataDog/datadog-agent/comp/core/workloadmeta"
+	"github.com/DataDog/datadog-agent/comp/dogstatsd/packets"
+	"github.com/DataDog/datadog-agent/pkg/tagset"
+	"github.com/DataDog/datadog-agent/pkg/telemetry"
+	"github.com/DataDog/datadog-agent/pkg/util/containers"
+	"github.com/DataDog/datadog-agent/pkg/util/containers/metrics"
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"go.uber.org/fx"
+)
+
+type dependencies struct {
+	fx.In
+
+	Lc      fx.Lifecycle
+	Config  configComponent.Component
+	Log     logComp.Component
+	Context context.Context
+	Wmeta   workloadmeta.Component `optional:"true"`
```

Comment:
Using [optional:"true"](https://pkg.go.dev/go.uber.org/fx#hdr-Optional_Dependencies) should be discouraged because it can hide errors. For example consider [this code](https://go.dev/play/p/xrzexhbCeif) where a user can expected to get an error because the dependency FooDep for CreateFoo is missing but the code doesn't raise an error. See also https://github.com/uber-go/fx/issues/613.